features/topology-map/plugins/org.opennms.features.topology.plugins.topo.linkd/src/main/java/org/opennms/features/topology/plugins/topo/linkd/internal/LinkdHopCriteria.java

```diff
@@ -35,61 +35,64 @@
 import org.opennms.features.topology.api.topo.DefaultVertexRef;
 import org.opennms.features.topology.api.topo.RefComparator;
 import org.opennms.features.topology.api.topo.VertexRef;
-import org.opennms.netmgt.topologies.service.api.OnmsTopology;
 
 /**
- * 
  * @author <a href=mailto:thedesloge@opennms.org>Donald Desloge</a>
  * @author <a href=mailto:seth@opennms.org>Seth Leger</a>
- *
  */
 public class LinkdHopCriteria extends VertexHopCriteria {
-    
-    public synchronized static VertexHopCriteria createCriteria(String nodeId, String nodeLabel) {
-        VertexHopCriteria criterion = new LinkdHopCriteria(nodeId, nodeLabel);
+
+    public static VertexHopCriteria createCriteria(String namespace, String nodeId, String nodeLabel) {
+        VertexHopCriteria criterion = new LinkdHopCriteria(namespace, nodeId, nodeLabel);
         return criterion;
     }
 
     private final String m_nodeId;
-	
-    private LinkdHopCriteria(String nodeId, String nodeLabel) {
-        super(nodeId,nodeLabel);
+    private final String m_namespace;
+
+    private LinkdHopCriteria(String namespace, String nodeId,
+            String nodeLabel) {
+        super(nodeId, nodeLabel);
         m_nodeId = nodeId;
+        m_namespace = namespace;
     }
-    
+
     @Override
     public String getNamespace() {
-        return OnmsTopology.TOPOLOGY_NAMESPACE_LINKD;
+        return m_namespace;
     }
 
     @Override
     public int hashCode() {
         final int prime = 31;
         int result = 1;
-        result = prime * result + ((m_nodeId == null) ? 0 : m_nodeId.hashCode());
+        result = prime * result
+                + ((m_nodeId == null) ? 0 : m_nodeId.hashCode());
         result = prime * result
                 + ((getNamespace() == null) ? 0 : getNamespace().hashCode());
         return result;
```

Comment:
If you have to touch this, please use `Objects.hash(m_nodeId, m_namespace)` instead.